src/player.cpp

```diff
@@ -2479,6 +2479,22 @@ void player::disp_morale()
     morale->display( ( calc_focus_equilibrium() - focus_pool ) / 100.0 );
 }
 
+time_duration player::estimate_effect_dur( const skill_id &relevant_skill,
+        const efftype_id &target_effect, const time_duration &error_magnitude,
+        int threshold, const Creature &target ) const
+{
+    int skill_lvl = get_skill_level( relevant_skill );
+
+    time_duration estimate = target.get_effect_dur( target_effect ) + rng( -1,
+                             1 ) * error_magnitude * rng( 0,
+                                     std::max( 0,
+                                             threshold - skill_lvl ) );
+    if( estimate < 0 ) {
+        estimate = 0;
+    }
```

Comment:
You can replace this by `time_duration estimate = std::max( 0, "formula" )`.

Reply:
I tried but it seems unhappy as formulat is `time_duration` and 0 is int and the output should be `time_duration`